src/EscapeAnalysis.jl

```diff
@@ -126,13 +127,15 @@ struct EscapeInfo
     Analyzed::Bool
     ReturnEscape::Bool
     ThrownEscape::LivenessSet
+    FinalizerEscape::Bool
```

Comment:
So how `FinalizerEscape` can be used? If we insert `finalize` call for allocations whose `FinalizerEscape` is `true`, we will end up inserting early `finalize` calls for anything escaping (since `AllEscape` also has `FinalizerEscape=true`.

Reply:
Should `FinalizerEscape` not be part of `AllEscape`?

My initial thoughts were that if nothing escaped other than the object being finalized, and the object being finalized does not otherwise escape, then we'd insert finalize at the end of the liveness range.